src/crawlee/basic_crawler/_basic_crawler.py

```diff
@@ -361,15 +361,10 @@ async def _get_proxy_info(self, request: Request, session: Session | None) -> Pr
             proxy_tier=None,
         )
 
-    async def get_request_provider(
-        self,
-        *,
-        id: str | None = None,
-        name: str | None = None,
-    ) -> RequestProvider:
+    async def get_request_provider(self) -> RequestProvider:
         """Return the configured request provider. If none is configured, open and return the default request queue."""
         if not self._request_provider:
-            self._request_provider = await RequestQueue.open(id=id, name=name, configuration=self._configuration)
+            self._request_provider = await RequestQueue.open(configuration=self._configuration)
```

Comment:
Why are we removing the `id` and `name` args?

Reply:
Because those only apply to `RequestQueue`, and `self._request_provider` may contain any `RequestProvider`. Also if you do `await crawler.get_request_provider(); await crawler.get_request_provider(name="Frankie")`, the second call won't return a Frankie, which is confusing.

Reply:
We could add a `get_request_queue` helper, but that may be even more confusing.

Reply:
Yeah, it makes sense. Either provide your own instance or you will get the default one (based on config).